src/tests/core/global/test_utxo_pool.c

```diff
@@ -1,161 +1,123 @@
 #include <stdio.h>
-#include "minunit.h"
+
+#include <setjmp.h>
+#include <cmocka.h>
+
 #include "utxo_pool.h"
 #include "crypto.h"
 #include "init_db.h"
+#include "fixtures_tx.h"
+#include "fixtures_global.h"
 
-int tests_run = 0;
-
-Transaction *_make_tx() {
-  Transaction *tx;
-  Input *in;
-  Output *out;
-  mbedtls_ecdsa_context *key_pair;
-
-  tx = malloc(sizeof(Transaction));
-  in = malloc(sizeof(Input));
-  out = malloc(sizeof(Output));
-
-  memset(in, 0, sizeof(Input));
-  key_pair = gen_keys();
-  in->pub_key = &(key_pair->private_Q);
-
-  memset(out, 0, sizeof(Output));
-
-  tx->num_inputs = 1;
-  tx->inputs = in;
-  tx->num_outputs = 1;
-  tx->outputs = out;
-
-  return tx;
-}
-
-static char *test_utxo_pool_init() {
-  utxo_pool_init_leveldb(TEST_DB_LOC);
-  mu_assert(
-    "UTXO DB is still null",
-    utxo_pool_db != NULL
-  );
-  mu_assert(
-    "UTXO path is null",
-    utxo_pool_path != NULL
-  );
-  destroy_db(&utxo_pool_db, utxo_pool_path);
-  return NULL;
+static void test_utxo_pool_init(void **state) {
+  (void)state;
+  assert_ptr_not_equal(utxo_pool_db, NULL);
```

Comment:
Think we may have discused that we should not check leveldb to Null, maybe we test the init funciton for a valid return ?